### DISCUSSION TITLE
2FA Web3 Wallet

### DISCUSSION THREAD

**Discussion by gamesms** [NillionNetwork/discussions#58] [Show and tell]
### Show and tell project type

Hackathon Project Submission

### Github Repo Link

https://github.com/gamesms/2faweb3wallet

### Project Description

Hi,

Many of us, including myself, have been victims of malware or other activities that compromise wallet secret phrases and drain our funds. With just one mistake—one single click in less than a second—you can lose your lifetime savings through crypto.

To address this, I have an idea: why not create a brand new wallet on the Nillion network that requires 2FA, similar to Ledger and Gmail, which use authenticator apps? This would prevent scammers and hackers from gaining access to funds, allowing people to be more secure and freely access their wallets. Please make this possible.

Thank you

### What problems does your project solve? How does it preserve privacy for users?

Problem: Wallet secret phrases can be compromised through unusual activities.
Solution: Implement 2FA to prevent this. Every transaction should require approval from a third-party app.

**Comment by anguschiu1** [comment]
We already have tons of wallets out there (just named a few)
![](https://blog.thirdweb.com/content/images/size/w1600/2023/01/The-Cryptocurrency-Wallet-Landscape-in-the-Blockchain-and-Web3-World.png)

With a wallet with 2FA just to use Nillion network would be one of the UX pain for users. We need 10x improvement to persuade people to switch wallets, which is not that simple.
